pkg/xmath/stats.go

```diff
@@ -0,0 +1,142 @@
+package xmath
+
+import (
+	"math"
+)
+
+type MADResult struct {
+	Mean float64
+	MAD  float64
+}
+
+// //
+//
+//  GENERIC FUNCTIONS
+//
+// //
+
+// FD computes the number of bins for a histogram using the Freedman-Diaconis
+// method applied to the given set of observations.
+func FD[V float64 | int64](values []V) float64 {
```

Comment:
its really cool to see this. i am generally curious how much we think we might use go in the future, and how potentially re-usable these little packages are as currently organized? like say we want to find IQR w go in some completely different context someday..